src/marqo/tensor_search/validation.py

```diff
@@ -1,38 +1,46 @@
 import json
-from enum import Enum
-from typing import Any, Dict, List, Optional, Type, Sequence, Union
-import semver
-
+from typing import Type, Sequence
 
 import jsonschema
 
 import marqo.core.models.marqo_index as marqo_index
-import marqo.exceptions as base_exceptions
+from marqo import marqo_docs
 from marqo.api.exceptions import (
     InvalidFieldNameError, InvalidArgError, InvalidDocumentIdError, DocTooLargeError)
+from marqo.core.models.marqo_index import *
 from marqo.tensor_search import constants as tensor_search_constants
 from marqo.tensor_search import enums, utils
 from marqo.tensor_search.enums import SearchMethod
+from marqo.tensor_search.models.custom_vector_object import CustomVector
 from marqo.tensor_search.models.delete_docs_objects import MqDeleteDocsRequest
-from marqo.tensor_search.models.search import SearchContext
 from marqo.tensor_search.models.mappings_object import (
     mappings_schema,
     multimodal_combination_mappings_schema,
     custom_vector_mappings_schema,
 )
-from marqo.core.models.marqo_index import *
-from marqo.tensor_search.models.custom_vector_object import CustomVector
-from marqo import marqo_docs
+from marqo.tensor_search.models.search import SearchContext
 
 
-def validate_query(q: Optional[Union[dict, str]], search_method: Union[str, SearchMethod]) -> Optional[Union[dict, str]]:
+def validate_query(q: Optional[Union[dict, str, CustomVector]], search_method: Union[str, SearchMethod]) -> Optional[
+    Union[dict, str, CustomVector]]:
     """
     Returns q if an error is not raised"""
-    usage_ref = "\nSee query reference here: https://docs.marqo.ai/0.0.13/API-Reference/search/#query-q"
+    usage_ref = f"\nSee query reference here: {marqo_docs.query_reference()}"
+
+    # TODO - it looks like API pydantic model is catching invalid input (e.g. bad dict) before it reaches this point
```

Comment:
Do we need to fix this? What's the expected behaviour?

Reply:
Not sure, asking @farshidz

Reply:
Yes, but I think it requires a fair bit of refactoring and also agreeing on where to do input validation

Reply:
can this release survive without it? It can be fast follow.